rust/agents/validator/src/submit.rs

```diff
@@ -99,6 +99,14 @@ impl ValidatorSubmitter {
                     message_id,
                 });
 
+                if checkpoint.index == correctness_checkpoint.index {
+                    // We got to the right height, now lets compare whether we got the right tree
+                    if checkpoint.root != correctness_checkpoint.root {
+                        // Bad news, bail
+                        bail!("Incorrect tree root, something went wrong");
+                    }
```

Comment:
should we put this before we push to the queue? or maybe share the `checkpoint == correctness_checkpoint` condition below somehow

Reply:
Since we bail anyways, i'm not sure I see the benefit of putting this in front of the queue push. Regarding condition sharing, the `checkpoint == correctness_checkpoint` is technically the else branch of the root equality check, but then imo we are further indenting the logic, so it felt easier to keep as is